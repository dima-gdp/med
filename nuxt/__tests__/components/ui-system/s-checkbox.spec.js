import { mount } from '@vue/test-utils'
import SCheckbox from '~/components/ui-system/s-checkbox'

describe('ui-system/s-checkbox', () => {
  const CHECKBOX_DEFAULT_PROPS = {
    value: false,
    disabled: false,
  }
  const checkboxFactory = (propsData = {}) => {
    return mount(SCheckbox, {
      propsData: {
        ...propsData,
      },
    })
  }

  it('s-checkbox отображается', () => {
    const wrapper = checkboxFactory(CHECKBOX_DEFAULT_PROPS)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('s-checkbox классы компонента при  дефолтных значения', () => {
    const wrapper = checkboxFactory(CHECKBOX_DEFAULT_PROPS)

    expect(wrapper.classes('s-checkbox')).toBe(true)
    expect(wrapper.classes('s-checkbox--checked')).toBe(false)
    expect(wrapper.classes('s-checkbox--disabled')).toBe(false)
  })

  it('s-checkbox при установленном значении disabled добавляется class (s-checkbox--disabled) и атрибут disabled', () => {
    const wrapper = checkboxFactory({ value: false, disabled: true })
    const checkbox = wrapper.find('input')
    expect(wrapper.classes('s-checkbox--disabled')).toBe(true)
    expect(checkbox.attributes('disabled')).toBeTruthy()
  })

  it('s-checkbox при клике изменяется значение на true|false, эммитятся значения, добовляется|убирается class (s-checkbox--checked)', async () => {
    const wrapper = checkboxFactory(CHECKBOX_DEFAULT_PROPS)

    const checkbox = wrapper.find('.s-checkbox')
    await checkbox.trigger('click')
    expect(wrapper.find('input').attributes('value')).toBe('true')
    expect(wrapper.emitted().input).toEqual([[true]])
    expect(wrapper.emitted().change).toEqual([[true]])
    await wrapper.vm.$nextTick()
    expect(wrapper.classes('s-checkbox--checked')).toBe(true)
  })
})
