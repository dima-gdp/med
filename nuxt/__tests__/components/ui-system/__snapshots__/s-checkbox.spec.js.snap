// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-system/s-checkbox s-checkbox отображается 1`] = `
<div class="s-checkbox">
  <div class="s-checkbox__box"><input type="checkbox" class="s-checkbox__input" value="false">
    <div class="s-checkbox__mask">
      <!---->
    </div>
  </div> <label for="" class="visually-hidden">

  </label>
</div>
`;
